multicast/recv.py

```
@@ -292,14 +369,13 @@ def joinstep(groups, port, iface=None, bind_group=None, isock=None):
 
 
 	"""
-	if not groups:
-		groups = []
+	groups, _, _, bind_group, _ = _validate_join_args(groups=groups, bind_group=bind_group)
 	if isock is None:
 		sock = multicast.genSocket()
 	else:
 		sock = isock.dup()
```

Ruff (0.8.2)

Use ternary operator `sock = multicast.genSocket() if isock is None else isock.dup()` instead of `if`-`else`-block

Replace `if`-`else`-block with `sock = multicast.genSocket() if isock is None else isock.dup()`

(SIM108)